pkg/apply/fieldmanager.go

```diff
@@ -47,11 +48,45 @@ func mergeManager(ctx context.Context, clusterClient client.ClusterClient, us *u
 			if err != nil {
 				return fmt.Errorf("failed to patch (type %s) for object %s %s: %v", patchType, objGVR.String(), us.GetName(), err)
 			}
+			// remove the old manager
```

Comment:
I dont understand why this line and the reset below is needed. 
The previous lines *should* remove the old manager.  
Can you paste here what the managed fields look like at this point following the patch?

Reply:
you are right. the changes you did to deprecate the old fieldmanager were enough and the extra I added
was overkill. However, your changes' code path wasn't executing because it was being called after the APPLY
and the APPLY is where the upgrade failed and got stuck. [Here](https://github.com/openshift/cluster-network-operator/pull/2136) is a PR that adds a GET before the APPLY and
if it sees an old manager, it will then execute your code path. I tested with it, and it works like we want.